simulation/traffic_simulator/src/entity/entity_manager.cpp

```diff
@@ -140,6 +106,16 @@ auto EntityManager::getEntityNames() const -> const std::vector<std::string>
   return names;
 }
 
+auto EntityManager::getEntity(const std::string & name) const
+  -> std::shared_ptr<traffic_simulator::entity::EntityBase>
+{
+  if (auto it = entities_.find(name); it != entities_.end()) {
+    return it->second;
+  } else {
+    return nullptr;
```

Comment:
@yamacir-kit 
Please take a look at the description for PR:

> Note: This method returns nullptr, due to the fact that the interpretation of the scenario operates in such a way that checking a condition, e.g. DistanceCondition, is called also for Entities that have not yet been spawned. For example, if for DistanceCondition any getEntity() returns nullptr, the condition returns a distance equal to NaN. For this reason, throwing an exception through getEntity() is not recommended.

In my first implementation, I developed an exception throw ([commit](https://github.com/tier4/scenario_simulator_v2/pull/1233/commits/70547e639122345c73fcb12f6a42a30f1924312e)). However, as a result of evaluations, I got many regressions (~150). For this reason, instead of catching an exception and developing a handler, I've decided to return `nullptr`. Otherwise, I would have had to add exception catching and its handling in many parts (there are many `getEntity()` calls in the  [simulator_core](https://github.com/tier4/scenario_simulator_v2/pull/1233/files#diff-a65d4236cdcc99de4a03ab886636dd35c20137227b9954579d05cea2e9a54f2e)). 
Therefore, I've added this note, because I also think it is a bit of a “workaround”.

Reply:
I missed the explanation. My apologies.

And I understand the situation. The code is fine as it is. However, please comment why this is not an exception throwing along with `return nullptr;`.